### PR TITLE
Collect filesystem metrics for the /run tmpfs mount point

### PR DESCRIPTION
**Pull request by istvanballok**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

Collect filesystem metrics for the `/run` tmpfs mount point

Previously, the filesystem metrics

// See https://github.com/prometheus/node_exporter/blob/68a6c78c0d1fd7359fa15d30b607b322eb512a8d/collector/filesystem_common.go#L134-L149

```
node_filesystem_files      Filesystem total file nodes.
node_filesystem_files_free Filesystem total free file nodes.
node_filesystem_readonly   Filesystem read-only status.
```

were only kept for the root `/` mount point.

With this change, they are kept for

```
mountpoint="/"
mountpoint="/run"
mountpoint="/tmp"
```

but are still discarded for the numerous workload related mount points,
that are not so relevant from the system component monitoring perspective.
e.g.

```
/var/lib/kubelet/pods/c585a9ce-04ad-4484-b61e-6dec922993c6/volumes/kubernetes.io~secret/kubeconfig
/run/containerd/io.containerd.runtime.v2.task/k8s.io/6335e3a8d7cb5542205cac258f3098fd7cebc454f765a6db9120670e2adc0e66/rootfs
```

Excluding these mount points helps to keep the cardinality of these metrics
low.

The new metrics are also allow listed so that they are persisted in prometheus.

A recording rule is added so that these metrics are federated by the aggregate-prometheus
in the garden namespace from all shoot control plane prometheus instances.

**Which issue(s) this PR fixes**:
Fixes #

These metrics can help to detect the issue when the `/run` tmpfs mount
runs out of inodes due to numerous empty directories "systemd-coredump"
in `/run/systemd/propagate` and hence e.g. containerd can not start new
containers.

**Special notes for your reviewer**:

**Release note**:

```other operator
The node-exporter is configured to collect filesystem metrics for the /run mount point.
```

### CODE REVIEW

**Comment by dguendisch** [comment]
Thank you!
/lgtm

File: charts/shoot-core/components/charts/monitoring/charts/node-exporter/templates/node-exporter.yaml [hunk 1]
```diff
@@ -83,7 +83,7 @@ spec:
         - --collector.diskstats
         - --collector.filefd
         - --collector.filesystem
-        - --collector.filesystem.mount-points-exclude=^/.+$
+        - --collector.filesystem.mount-points-exclude=^/(run|var)/.+$|^/(boot|dev|sys|usr)($|/.+$)
```
Comment (acumino) [inline charts/shoot-core/components/charts/monitoring/charts/node-exporter/templates/node-exporter.yaml:86]: If I understand this regex correctly, we want to include `/run` and exclude everything i.e. `/run/*` ?

Comment (istvanballok) [reply charts/shoot-core/components/charts/monitoring/charts/node-exporter/templates/node-exporter.yaml:86]: Exactly.
There are many subpaths (for each container) that we'd like to ignore: `/run/containerd/io.containerd.runtime.v2.task/k8s.io/6335e3a8d7cb5542205cac258f3098fd7cebc454f765a6db9120670e2adc0e66/rootf`.

Motivation: the `/run` mount can fill up with empty directories. See https://github.tools.sap/kubernetes/landscape-issues/issues/62

**Review by acumino — COMMENTED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]

**Pull request merged**